apps/guide/src/content/04-popular-topics/03-collectors.mdx

```diff
@@ -0,0 +1,223 @@
+
+try {
+	const collected = await interaction.channel.awaitMessages({
+		filter: collectorFilter,
+		max: 1,
+		time: 30000,
+		errors: ['time'],
+	});
+
+	await interaction.followUp(`${collected.first().author} got the correct answer!`);
+} catch {
+	await interaction.followUp('Looks like nobody got the answer this time.');
+}
+```
+
+</CH.Code>
+
+<Alert title="Tip" type="info">
+	If you don't understand how _`.some()`_ works, you can read about it in more detail
+	[here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).
+</Alert>
+
+In this filter, you iterate through the answers to find what you want. You would like to ignore the case because simple typos can happen, so you convert each answer to its lowercase form and check if it's equal to the response in lowercase form as well. In the options section, you only want to allow one answer to pass through, hence the _`max: 1`_ setting.
+
+The filter looks for messages that match one of the answers in the array of possible answers to pass through the collector. The options (the second parameter) specifies that only a maximum of one message can go through the filter successfully before the Promise successfully resolves. The errors section specifies that time will cause it to error out, which will cause the Promise to reject if one correct answer is not received within the time limit of one minute. As you can see, there is no _`collect`_ event, so you are limited in that regard.
+
+## Reaction collectors
+
+### Basic reaction collector
+
+These work quite similarly to message collectors, except that you apply them on a message rather than a channel. This example uses the <DocsLink type="class" parent="Message" symbol="createReactionCollector" brackets /> method. The filter will check for the 👍 emoji–in the default skin tone specifically, so be wary of that. It will also check that the person who reacted shares the same id as the author of the original message that the collector was assigned to.
+
+```js
+const collectorFilter = (reaction, user) => {
+	return reaction.emoji.name === '👍' && user.id === message.author.id;
+};
+
+const collector = message.createReactionCollector({ filter: collectorFilter, time: 15000 });
+
+collector.on('collect', (reaction, user) => {
+	console.log(`Collected ${reaction.emoji.name} from ${user.tag}`);
+});
+
+collector.on('end', (collected) => {
+	console.log(`Collected ${collected.size} items`);
+});
+```
+
+### Await reactions
+
+<DocsLink type="class" parent="Message" symbol="awaitReactions" brackets /> works almost the same as a reaction collector,
+except it is Promise-based. The same differences apply as with channel collectors.
+
+```js
+const collectorFilter = (reaction, user) => {
+	return reaction.emoji.name === '👍' && user.id === message.author.id;
+};
+
+try {
+	const collected = await message.awaitReactions({ filter: collectorFilter, max: 4, time: 60000, errors: ['time'] });
```

Comment:
```suggestion
	const collected = await message.awaitReactions({ filter: collectorFilter, max: 4, time: 60_000, errors: ['time'] });
```

Reply:
Also I will add that `max` makes no sense here, since the filter will only match the author of the message, so it will always be 1 or 0. Maybe modify the filter instead 👀

Reply:
doesn't it only mean the user can react up to 4 times? (even if it's the same reaction)

Reply:
also, might a dumb question but, isn't the message author the client user? like we're doing `.awaitMessages` on the bot's message where the reaction is
so `user.id` won't equal `message.author.id` no?

Reply:
> doesn't it only mean the user can react up to 4 times? (even if it's the same reaction)

That would mean the same user reacts, unreacts, reacts, unreacts, double what I just said 👀

Reply:
> also, might a dumb question but, isn't the message author the client user? like we're doing `.awaitMessages` on the bot's message where the reaction is so `user.id` won't equal `message.author.id` no?

It's not really given as a context if its the bot's response or the awaiting happens on the message sent by a user, so it's up for interpretation. But if it is the bot's message, then yes